Sources/Layers/Concat.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Concat.swift
+//  Bender
+//
+//  Created by Diego Ernst on 6/27/17.
+//
+//
+
+import MetalPerformanceShaders
+
+open class Concat: NetworkLayer {
+
+    var pipeline: MTLComputePipelineState!
+    let axis: LayerSizeAxis
+
+    let maxInputTextures = 10 // must match kernel max input textures
+
+    public init(axis: LayerSizeAxis, id: String? = nil) {
+        self.axis = axis
+        super.init(id: id)
+    }
+
+    open override func initialize(network: Network, device: MTLDevice) {
+        super.initialize(network: network, device: device)
+        let incoming = getIncoming()
+
+        // correctness checks
+
+        // TODO: implement shader to support concat along z without restricting it to multiples of 4
+
+        assert(axis != .f || incoming.reduce(true) { $0.0 && ($0.1.outputSize.f % 4 == 0) }, "Concat: all z dimensions must be multiple of 4")
+        assert(!incoming.isEmpty, "Concat: expects at least one inputs")
+        assert(incoming.count <= maxInputTextures, "Concat: only accepts \(maxInputTextures) incomming nodes at most")
+
+        let allInputTexturesWithMoreThan4Channels = incoming.reduce(true) { $0.0 && $0.1.outputSize.f > 4 }
+        let allInputTexturesWithLessThanOrEqualTo4Channels = incoming.reduce(true) { $0.0 && $0.1.outputSize.f <= 4 }
+        assert(allInputTexturesWithMoreThan4Channels || allInputTexturesWithLessThanOrEqualTo4Channels, "All z dimensions must be either > 4 or <= 4 at the same time")
+
+        let axisThatMustBeEqual = LayerSizeAxis.all.filter { $0 != axis }
+        var axisValues = [LayerSizeAxis: Int]()
+
+        let sampleSize: LayerSize! = incoming[0].outputSize
+        axisThatMustBeEqual.forEach { axisValues[$0] = sampleSize[$0] }
```

Comment:
Mmm not sure, `axisValues` is actually a dictionary containing for each `axis` belonging to `axisThatMustBeEqual` its corresponding Int value